gossipd/gossipd.c

```diff
@@ -198,6 +199,21 @@ static bool get_node_announcement(const tal_t *ctx,
 	}
 
 	*wireaddrs = read_addresses(ctx, addresses);
+
+	/* Read out liquidity ad */
+	if (na_tlv->option_will_fund) {
+		*ad = tal(ctx, struct liquidity_ad);
+		(*ad)->lease_basis =
+			na_tlv->option_will_fund->funding_fee_proportional_basis;
+		(*ad)->lease_base_sat =
+			amount_sat(na_tlv->option_will_fund->funding_fee_base_sat);
+		(*ad)->channel_fee_basis =
+			na_tlv->option_will_fund->channel_fee_proportional_basis;
+		(*ad)->channel_base_msat =
+			amount_msat(na_tlv->option_will_fund->channel_fee_base_msat);
```

Comment:
This is getting messy.  Can we add a subtype to the spec and use that everywhere?